### ISSUE TITLE
MinGW-w64 6.3.0 compilation error

### ISSUE THREAD

**Issue by chronoxor** [chriskohlhoff/asio#186]
I found a compilation error under MinGW-w64 6.3.0:

C:/projects/cppserver/modules/asio/asio/include/asio/detail/impl/win_iocp_socket_service_base.ipp:748:27: error: invalid conversion from 'FARPROC {aka long long int (*)()}' to 'void*' [-fpermissive]
       ptr = GetProcAddress(h, "NtSetInformationFile");
             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~

which is reproducing after 

Revision: d3ca90a05c56f6b90a2353a84f57ecee9a0a6c42
Author: Christopher Kohlhoff <chris@kohlhoff.com>
Date: 22.02.2017 14:46:15
Message: Add ability to release ownership of the underlying native socket.

**Comment by chronoxor** [comment]
https://github.com/chriskohlhoff/asio/pull/187

**Comment by chriskohlhoff** [comment]
Fixed on master in 3d5367da14a8880161bde8a1929a1b8c40968668.
